src/components/NavComp.jsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import {Navbar, NavbarBrand, NavbarContent, NavbarItem, NavbarMenuToggle, NavbarMenu, NavbarMenuItem, Link,Button, Switch} from "@nextui-org/react";
+import { MoonIcon } from "../assets/MoonIcon";
+import { SunIcon } from "../assets/SunIcon";
+
+
+export default function NavComp({ toggleDarkMode }) {
+  const [isMenuOpen, setIsMenuOpen] = React.useState(false);
+
+  const menuItems = [
```

Comment:
The `menuItems` and `items` variables should be separated into their own file(s) to keep the code clean and modular. 

For the navigation items, consider creating an object that holds essential information, such as redirection paths, class names, labels, and other relevant properties. You can then iterate over this array in the component to render the navigation items dynamically. This approach will make your navbar component more scalable and easier to maintain as it grows.